mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,222 @@
+        y : array, shape ([n_epochs], n_times)
+            The output feature for the model.
+        """
+        if X.ndim == 2:
+            # Ensure we have a 3D input
+            X = X[np.newaxis, ...]
+        if X.shape[-1] != y.shape[-1]:
+            raise ValueError('X any y do not have the same n_times\n'
+                             '%s != %s' % (X.shape[-1], y.shape[-1]))
+        n_epochs, n_feats, n_times = X.shape
+        X_del, msk = self._delay_for_fit(X)
+
+        # Reshape so we have times x features
+        X_del = X_del[..., ~msk]
+        y = y[..., ~msk]
+
+        # Convert to 2d by making epochs 1st axis and hstacking
+        X_del = X_del.swapaxes(0, 1).reshape(n_epochs, -1, X_del.shape[-1])
+        X_del = np.hstack(X_del)
+        y = np.hstack(y)
+
+        # Update feature names if we have none
+        if self.feature_names is None:
+            self.feature_names = ['feature_%s' % ii for ii in range(n_feats)]
+        if len(self.feature_names) != n_feats:
+            raise ValueError('n_features in X does not match feature names '
+                             '(%s != %s)' % (n_feats, len(self.feature_names)))
+
+        self.model.fit(X_del.T, y.T)
+        self.mask_remove = msk
+
+        coefs = get_coefs(self.model)
+        self.coef_ = coefs.reshape([len(self.feature_names), len(self.delays)])
+
+    def predict(self, X, y=None):
+        """Make predictions using a receptive field.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
+        y : None
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        y_pred : array, shape (n_times * n_epochs)
+            The output predictions with time concatenated.
+        """
+        X_del, msk = self._delay_for_fit(X)
+        X_del = X_del[..., ~msk]
+        X_del = X_del.reshape([-1, X_del.shape[-1]]).T
+        y_pred = self.model.predict(X_del)
+        self.mask_pred = ~msk
+        return y_pred
+
+    def _delay_for_fit(self, X):
+        # First delay
+        X_del = delay_time_series(X, self.tmin, self.tmax, self.sfreq)
+
+        # Mask for removing edges later
+        msk_helper = np.ones(X.shape[-1])
+        msk_helper = delay_time_series(msk_helper, self.tmin,
+                                       self.tmax, self.sfreq)
+        msk = np.any(msk_helper == 0, axis=0)
+        return X_del, msk
+
+
+def delay_time_series(X, tmin, tmax, sfreq, newaxis=0):
+    """Return a time-lagged input time series.
+
+    Parameters
+    ----------
+    X : array, shape ([n_epochs], n_features, n_times)
+        The time series to delay.
+    tmin : int | float
+        The tmining lag. Negative values correspond to times in the past.
+    tmax : int | float
+        The ending lag. Positive values correspond to times in the future.
+        Must be >= tmin.
+    sfreq : int | float
```

Comment:
I don't believe it will throw an error if it's a float, but maybe for the docs better to call it an int